### PR TITLE
[ZkTracer] test: add test for signed SMOD

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**
This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> **Low Risk**
> Test-only changes that add coverage for a signed modulo edge case; no production logic is modified.
> 
> **Overview**
> Extends `ModTest` to cover an additional `SMOD` scenario where the numerator is positive and the denominator is negative, using explicit 256-bit values.
> 
> Also cleans up the test bytecode construction by statically importing `SMOD`/`POP` and adding `Bytes` usage for hex literals.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0c8a3e44f09ebda19c16aa704359cc69a2faaf50. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by lorenzogentile404 — APPROVED** [review]
LGTM

File: tracer/arithmetization/src/test/java/net/consensys/linea/zktracer/module/mod/ModTest.java [hunk 1]
```diff
@@ -55,6 +59,16 @@ void testSignedSmod(TestInfo testInfo) {
...
+                // SMOD with num > 0 and denominator < 0
+                .push(
+                    Bytes.fromHexString(
+                        "0x8000000000000000000000000000000000000000000000000000000000001111")) // denom
+                .push(
+                    Bytes.fromHexString(
+                        "0x0000000000000000000000000000000000000000000000000000000011111111")) // num
+                .op(SMOD)
+                .op(POP)
                 .compile())
```
Comment (lorenzogentile404) [inline tracer/arithmetization/src/test/java/net/consensys/linea/zktracer/module/mod/ModTest.java:72]: It may be worth splitting the different scenario into separated small tests.

**Review by OlivierBBB — APPROVED** [review]
